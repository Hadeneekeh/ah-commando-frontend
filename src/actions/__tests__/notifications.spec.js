import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  getNotifications, markAsRead, markAllRead, updateSubscription,
} from '../notifications';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-haven-staging.herokuapp.com/api/v1';

describe('Home Page Articles Actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });

  afterEach(() => {
    nock.cleanAll();
  });

  const response = {
    data: [
      {
        title: 'Example title 1',
        body: 'Some test',
      },
      {
        title: 'Example title 2',
        body: 'Some test',
      },
      {
        title: 'Example title 3',
        body: 'Some test',
      },
    ],
  };

  it('Should get all user Notifications', () => {
    nock(url)
      .get('/notifications/')
      .reply(200, response);

    return store.dispatch(getNotifications()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('Should Mark a notification as read', () => {
    nock(url)
      .patch('/notifications/2/read')
      .reply(200, response);

    return store.dispatch(markAsRead(2)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });


  it('Should mark All notifications as read', () => {
    nock(url)
      .patch('/notifications/read')
      .reply(200, response);

    return store.dispatch(markAllRead()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('Should subcribe to email notifications', () => {
    nock(url)
      .patch('/notifications/email/subscribe')
      .reply(200, response);

    return store.dispatch(updateSubscription(true)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('Should unsubscribe from email notifications', () => {
    nock(url)
      .delete('/notifications/email/subscribe')
      .reply(200, response);

    return store.dispatch(updateSubscription(false)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
