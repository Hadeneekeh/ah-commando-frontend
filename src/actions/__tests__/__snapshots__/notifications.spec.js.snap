// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Articles Actions Should Mark a notification as read 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "UPDATE_NOTIFICATIONS",
  },
]
`;

exports[`Home Page Articles Actions Should get all user Notifications 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "FETCH_NOTIFICATIONS",
  },
]
`;

exports[`Home Page Articles Actions Should mark All notifications as read 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "MARK_READ",
  },
]
`;

exports[`Home Page Articles Actions Should subcribe to email notifications 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "UPDATE_SUB",
  },
]
`;

exports[`Home Page Articles Actions Should unsubscribe from email notifications 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "Some test",
          "title": "Example title 1",
        },
        Object {
          "body": "Some test",
          "title": "Example title 2",
        },
        Object {
          "body": "Some test",
          "title": "Example title 3",
        },
      ],
    },
    "type": "UPDATE_SUB",
  },
]
`;
